tests/validate/ps_aie_test/src/host.cpp

```diff
@@ -0,0 +1,143 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License"). You may
+* not use this file except in compliance with the License. A copy of the
+* License is located at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations
+* under the License.
+*/
+#include <boost/filesystem.hpp>
+#include <algorithm>
+#include <vector>
+// XRT includes
+#include "experimental/xrt_system.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_kernel.h"
+#include "xrt/xrt_bo.h"
+
+#define WIDTH 8
+#define HEIGHT 8 
+#define SIZE (WIDTH * HEIGHT)
+
+static void printHelp() {
+    std::cout << "usage: %s <options>\n";
+    std::cout << "  -p <path>\n";
+    std::cout << "  -d <device> \n";
+    std::cout << "  -s <supported>\n";
+    std::cout << "  -h <help>\n";
+}
+
+int main(int argc, char** argv) {
+    std::string dev_id = "0";
+    std::string test_path;
+    std::string b_file = "/ps_aie.xclbin";
+    bool flag_s = false;
+    for (int i = 1; i < argc; i++) {
+        if ((strcmp(argv[i], "-p") == 0) || (strcmp(argv[i], "--path") == 0)) {
+            test_path = argv[i + 1];
+        } else if ((strcmp(argv[i], "-d") == 0) || (strcmp(argv[i], "--device") == 0)) {
+            dev_id = argv[i + 1];
+        } else if ((strcmp(argv[i], "-s") == 0) || (strcmp(argv[i], "--supported") == 0)) {
+            flag_s = true;
+        } else if ((strcmp(argv[i], "-h") == 0) || (strcmp(argv[i], "--help") == 0)) {
+            printHelp();
+            return 1;
+        }
+    }
+
+    if (test_path.empty()) {
+        std::cout << "ERROR : please provide the platform test path to -p option\n";
+        return EXIT_FAILURE;
+    }
+    auto binaryFile = boost::filesystem::path(test_path) / b_file;
+    std::ifstream infile(binaryFile.string());
+    if (flag_s) {
+        if (!infile.good()) {
+            std::cout << "\nNOT SUPPORTED" << std::endl;
+            return EOPNOTSUPP;
+        } else {
+            std::cout << "\nSUPPORTED" << std::endl;
+            return EXIT_SUCCESS;
+        }
+    }
+
+    if (!infile.good()) {
+        std::cout << "\nNOT SUPPORTED" << std::endl;
+        return EOPNOTSUPP;
+    }
+
+    auto num_devices = xrt::system::enumerate_devices();
+
+    xrt::device device;
+    auto pos = dev_id.find(":");
+    if (pos == std::string::npos) {
+        uint32_t device_index = stoi(dev_id);
+        if (device_index >= num_devices) {
+            std::cout << "The device_index provided using -d flag is outside the range of "
+                         "available devices\n";
+            return EXIT_FAILURE;
+        }
+        device = xrt::device(device_index);
+    } else {
+        device = xrt::device(dev_id);
+    }
```

Comment:
Bit mysterious to me.  What is it doing?
Why not just 
`xrt::device device {dev_id};`
The device constructor supports string to identify device per
```
 /**
   * device() - Constructor from string
   *
   * @param bdf
   *  String identifying the device to open.
   *
   * If the string is in BDF format it matched against devices
   * installed on the system.  Otherwise the string is assumed
   * to be a device index.
   *
   * Throws if string format is invalid or no matching device is
   * found.
   */
  XCL_DRIVER_DLLESPEC
  explicit
  device(const std::string& bdf);
```
So you don't need to enumerate devices or to parse the devid.